### PR TITLE
frdrpc: rename proto, use Docker for compilation, move to GitHub CI

### PR DESCRIPTION
**Pull request by guggero**
The main purpose of the PR is to rename `rpc.proto` to `faraday.proto` in an attempt to get rid of the error `WARNING: proto: file "rpc.proto" is already registered` we see wherever we use both Faraday as well as `lnd` as libraries.

While we're at it, we also use Docker for compiling the RPC protos, update the golang version and move the CI pipeline to GitHub.

### CODE REVIEW

File: frdrpc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  protobuf-compiler='3.6.1*' \
+  clang-format='1:7.0*'
+
+# We don't want any default values for these variables to make sure they're
+# explicitly provided by parsing the go.mod file. Otherwise we might forget to
+# update them here if we bump the versions.
+ARG PROTOC_GEN_VERSION
+ARG GRPC_GATEWAY_VERSION
+
+ENV FALAFEL_VERSION="v0.7.1"
```
Comment (carlaKC) [inline frdrpc/Dockerfile:14]: `FALAFEL_VERSION` unnecessary?

File: go.mod [hunk 2]
```diff
@@ -21,4 +21,4 @@ require (
 	gopkg.in/macaroon.v2 v2.1.0
 )
 
-go 1.13
+go 1.15
```
Comment (carlaKC) [inline go.mod:24]: nit: commit message doesn't quite fit, since we're bumping min version and docker versions to different values

**Review by carlaKC — APPROVED** [review]
Great improvement! Just some nits.

File: frdrpc/Dockerfile (hunk 1 above)
Comment (guggero) [reply frdrpc/Dockerfile]: Correct, obviously this isn't needed :sweat_smile:

**Pull request merged**